formats/gltf/model.go

```diff
@@ -51,3 +51,86 @@ type PolyformTexture struct {
 	URI     string
 	Sampler *Sampler
 }
+
+func (pm PolyformMaterial) equal(other PolyformMaterial) bool {
+	if pm.Name != other.Name {
+		return false
+	}
+	if !pm.PbrMetallicRoughness.equal(other.PbrMetallicRoughness) {
+		return false
+	}
+	if !colorsEqual(pm.EmissiveFactor, other.EmissiveFactor) {
+		return false
+	}
+
+	if (pm.AlphaMode == nil) != (other.AlphaMode == nil) {
+		return false
+	} else if pm.AlphaMode != nil && other.AlphaMode != nil && *pm.AlphaMode != *other.AlphaMode {
+		return false
+	}
+
+	if !float64PtrsEqual(pm.AlphaCutoff, other.AlphaCutoff) {
+		return false
+	}
+	if len(pm.Extensions) != len(other.Extensions) {
+		return false
+	}
+	for i, ext := range pm.Extensions {
+		if i >= len(other.Extensions) || ext != other.Extensions[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (pt *PolyformTexture) equal(other *PolyformTexture) bool {
+	if (pt == nil) != (other == nil) {
+		return false
+	} else if pt == nil {
+		return true
+	}
+	return pt.URI == other.URI
+}
+
+func (pmr *PolyformPbrMetallicRoughness) equal(other *PolyformPbrMetallicRoughness) bool {
+	if (pmr == nil) != (other == nil) {
+		return false
+	}
+	if pmr == nil {
+		return true
+	}
```

Comment:
Oh, that's an obscure one, yeah :)

It took some effort for me to figure out what Claude generated here, but logically it's correct.
Previous check looks at `(pmr == nil) != (other == nil)`, i.e. it will trigger if only one of `pmr` or `other` is not nil. It will not trigger if both are not nil or both are nil. This check looks if `pmr == nil`, but since previous hasn't triggered - this effectively looks for `pmr == nil && other == nil`.  In this case - they are equal. 
 
It's a neat trick, but takes unnecessary amount of braincycles to understand for the worth of removing one check, this microoptimisation isn't worth it, I'll make it more explicit and readable.

This pattern is used in few other places as well.